src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,148 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int BASIC_LENGTH = 10;
+    private static final int LENGTH_STEP = 5;
+    private T[] list;
+    private int index;
+
+    public ArrayList() {
+        list = (T[]) new Object[BASIC_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        if (index == list.length) {
+            addSpase();
+        }
+        list[index] = value;
+        index++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index >= 0) {
+            if (index > this.index) {
+                /* If index of element doesn`t fall into the array */
+                throw new ArrayListIndexOutOfBoundsException("Incorrect index of list");
```

Comment:
Better use
```suggestion
            if (index > this.index || index < 0) {
                throw new ArrayListIndexOutOfBoundsException("Incorrect index of list");
```
But this check is repeated in several ways. Better practice writing a separate method for it
And better don't use "this" keyword